app/controllers.py

```diff
@@ -0,0 +1,135 @@
+import random
+import string
+from datetime import datetime
+
+from flask import session, redirect, request, url_for
+from flask.json import dump
+
+from app import db, bp
+from app.models import ZdravotnickeStredisko, PrakticiLogin, PrakticiKapacity, Vakcina
+
+
+@bp.route("/praktici_admin/register", methods = ['POST'])
+def praktici_admin_register():
+    session.pop('error', None)
+    session.pop('success', None)
+
+    id = request.form.get('id')
+
+    zarizeni = db.session.query(ZdravotnickeStredisko) \
+        .filter(ZdravotnickeStredisko.zdravotnicke_zarizeni_kod == id) \
+        .one_or_none()
+
+    user = db.session.query(PrakticiLogin) \
+        .filter(PrakticiLogin.zdravotnicke_zarizeni_kod == id) \
+        .one_or_none()
+
+    if zarizeni is None:
+        session['error'] = 'Zdravotnické zařízení se zadaným kódem neexistuje.'
+    elif user is not None:
+        session['error'] = 'Zdravotnické zařízení je už registrováno.'
+    else:
+        random.seed(datetime.now())
+        characters = string.ascii_letters + string.digits
+        passwd = ''.join(random.choice(characters) for i in range(8))
+
+        user = PrakticiLogin(id, passwd)
+        db.session.add(user)
+
+        vaccines = db.session.query(Vakcina).all()
+
+        for vaccine in vaccines:
+            free_vaccine = PrakticiKapacity()
+            free_vaccine.zdravotnicke_zarizeni_kod = id
+            free_vaccine.typ_vakciny = vaccine.vyrobce
+            free_vaccine.kraj = zarizeni.kraj
+            free_vaccine.mesto = zarizeni.obec
+            free_vaccine.nazev_ordinace = zarizeni.nazev_cely
+            free_vaccine.adresa = f'{zarizeni.ulice} {zarizeni.cislo_domu}, {zarizeni.psc} {zarizeni.obec}'
+            free_vaccine.pocet_davek = 0
+            free_vaccine.kontakt_email = zarizeni.email
+            free_vaccine.kontakt_tel = zarizeni.telefon
+            free_vaccine.poznamka = ''
+            db.session.add(free_vaccine)
+
+        db.session.commit()
+
+        session['user_id'] = id
+        session['user_passwd'] = passwd
+
+        session['success'] = f"Děkujeme za registraci. Pro ordinaci '{id}' je Vaše heslo '{passwd}'."
+
+    return redirect(url_for('view.praktici_admin'))
+
+
+@bp.route("/praktici_admin/login", methods = ['POST'])
+def praktici_admin_login():
+    session.pop('error', None)
+    session.pop('success', None)
+
+    id = request.form.get('id')
+    passwd = request.form.get('passwd')
+
+    user = db.session.query(PrakticiLogin) \
```

Comment:
Fetch nazev_cely from zdravotnicke_stredisko -> to show this information

Reply:
it's in another table, where to show it?